src/v/crypto/crypto.cc

```diff
@@ -46,6 +51,79 @@ std::ostream& operator<<(std::ostream& os, format_type type) {
 }
 
 namespace internal {
+bool uses_default_provider(digest_type type) {
+    // The following set contains digests that are not present in the OpenSSL
+    // FIPS module however their use is not disallowed as long as the operation
+    // being performed is not security relevant (e.g. performing a checksum
+    // rather than an HMAC)
+    static const absl::flat_hash_set<digest_type> non_fips_digest_types{
+      digest_type::MD5};
+
+    return non_fips_digest_types.contains(type);
+}
+bool uses_default_provider(const EVP_MD* md) {
+    bool digest_type_set = false;
+    digest_type type;
+    struct callback_func_data {
+        bool& digest_type_set;
+        digest_type& type;
+    };
```

Comment:
question: what is the purpose of the extra indirection? it looks like `data` and its constituents should have the same lifetime?

Reply:
good point!